### PR TITLE
Public roadmap for the project

### PR DESCRIPTION
**Pull request by gouthamve**
Quickly jotted this down, as a follow up to https://github.com/cncf/toc/pull/315#issuecomment-651376899

Once I have consensus that these are the features we actually want and after incorporating more suggestions, I'll tighten up the language.

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+The following is only a selection of some of the major features we plan to implement in the near future. To get a more complete overview of planned features and current work, see the issue trackers for the various repositories, for example, the [Cortex repo](https://github.com/cortexproject/cortex/issues).
+
+## Making a NoSQL store optional
+
+Currently Cortex depends on a NoSQL store (like Bigtable, DynamoDB, Cassandra) for its index storage. This adds a lot of complexity and cost to manage and we are currently working on making the NoSQL store optional by using an ObjectStore for index storage as well.
+
+## helm charts and other packaging
+
+We don't provide a set of helm charts and dashboards / alerts to our users yet. This is one of the most requested features and something we will tackle in the immediate future. We also plan on publishing debs, rpms along with guides on how to run Cortex on bare-metal.
```
Comment (tomwilkie) [inline docs/roadmap.md:16]: we have helm chart here: https://github.com/cortexproject/cortex-helm-chart so I'd make this more about being first class citizen / better quality / more supported.

**Comment by tomwilkie** [comment]
Some ideas:
- Ruler in single binary
- Horizontally scalable / shared alertmanager
- Groupcache so we don't need memcache / better caching
- More reloadable config
- Expose query performance metadata to Grafana explore
- Persist Prometheus metadata
- Exemplars
- Import data

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+slug: roadmap 
+---
+
+The following is only a selection of some of the major features we plan to implement in the near future. To get a more complete overview of planned features and current work, see the issue trackers for the various repositories, for example, the [Cortex repo](https://github.com/cortexproject/cortex/issues).
+
+## Making a NoSQL store optional
+
+Currently Cortex depends on a NoSQL store (like Bigtable, DynamoDB, Cassandra) for its index storage. This adds a lot of complexity and cost to manage and we are currently working on making the NoSQL store optional by using an ObjectStore for index storage as well.
+
+## helm charts and other packaging
```
Comment (pracucci) [inline docs/roadmap.md:14]:
```suggestion
## Helm charts and other packaging
```

File: docs/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+---
+title: "Roadmap"
+linkTitle: "Roadmap"
+weight: 50
+slug: roadmap 
+---
+
+The following is only a selection of some of the major features we plan to implement in the near future. To get a more complete overview of planned features and current work, see the issue trackers for the various repositories, for example, the [Cortex repo](https://github.com/cortexproject/cortex/issues).
```
Comment (weeco) [inline docs/roadmap.md:8]: Are the roadmap items ordered by priority?

File: docs/roadmap.md (hunk 1 above)
Comment (khaines) [reply docs/roadmap.md:16]: +1 to getting more involvement in the helm chart. Community is putting in PRs but I'm the only reviewer of late and my time lately has been very sporadic

File: docs/roadmap.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## Soft Multitenancy
+
+Currently our multitenancy allows a tenant to view _all_ their metrics and only their metrics. There is no way for an "admin" tenant to view all the metrics in the system but for particular teams to only view theirs. This is another feature we plan to add into Cortex.
+
+## Exemplar and Prometheus metadata support
+
+There is a currently an effort in Prometheus to add [exemplar support](https://docs.google.com/document/d/1ymZlc9yuTj8GvZyKz1r3KDRrhaOjZ1W1qZVW_5Gj7gA/edit) and we should be an active stakeholder in the discussion. The plan is to propagate the exemplars through remote write and make them available for querying in Cortex. We currently have experimental metadata support for Prometheus but this is using the Grafana Cloud Agent. We should help move this PR forward and also add persistence of the metadata (right now it's only in-mem).
```
Comment (pracucci) [inline docs/roadmap.md:44]:
```suggestion
There is currently an ongoing effort in Prometheus to add [exemplar support](https://docs.google.com/document/d/1ymZlc9yuTj8GvZyKz1r3KDRrhaOjZ1W1qZVW_5Gj7gA/edit) and we should be an active stakeholder in the discussion. The plan is to propagate the exemplars through remote write and make them available for querying in Cortex. We currently have experimental metadata support for Prometheus but this is using the Grafana Cloud Agent. We should help move this PR forward and also add persistence of the metadata (right now it's only in-mem).
```

**Pull request merged**